### PR TITLE
Enabling Third-party Smart Contract Account Applications on Polkadot Based on Account Abstraction.md

### PR DESCRIPTION
**Pull request by mathida520**
**Create Enabling Third-party Smart Contract Account Applications on Polkadot Based on Account Abstraction.md**

#### Project Abstract

Account abstraction upgrades the blockchain accounts of users to smart contracts, but it's hard for users to write and deploy smart contracts by themselves. To help users better set and use their smart contract accounts easily under AA, we would like to extend the existing AA standard to enable all smart contract developers to provide their third-party account applications and services for users. They can greatly improve the user experience, capacity and wallet innovation of AA via our standard interface without compromising on decentralization, self-sovereign and censorship-resistance.

#### Grant [level](https://github.com/w3f/Grants-Program#level_slider-levels)

- [ ] **Level 1**:  Up to $10,000, 2 approvals
- [x] **Level 2**:  Up to $30,000, 3 approvals
- [ ] **Level 3**:  Unlimited, 5 approvals (for >$100k: Web3 Foundation Council approval)

#### Application Checklist

- [x] The [application template](https://github.com/w3f/Grants-Program/blob/master/applications/application-template.md) has been copied and aptly renamed (`project_name.md`).
- [x] I have read the [application guidelines](https://github.com/w3f/Grants-Program/blob/master/docs/Support%20Docs/grant_guidelines_per_category.md).
- [x] Payment details have been provided (bank details via email _or_ BTC, Ethereum (USDC/DAI) or Polkadot/Kusama (USDT) address in the application).
- [x] The software delivered for this grant will be released under an open-source license specified in the application.
- [x] The initial PR contains only one commit (squash and force-push if needed).
- [x] The grant will only be announced once the first milestone [has been accepted](https://github.com/w3f/Grant-Milestone-Delivery#process) (see the [announcement guidelines](https://github.com/w3f/Grants-Program/blob/master/docs/Support%20Docs/announcement-guidelines.md)).
- [ ] I prefer the discussion of this application to take place in a private Element/Matrix channel. My username is: `@_______:matrix.org` (change the homeserver if you use a different one)

### CODE REVIEW

**Comment by mathida520** [comment]
> Thanks for your interest in our grants program. But this application currently seems mostly focused on Ethereum and the account types there. Are you aware that Polkadot already has an account abstraction in the form of proxies, see for example, https://twitter.com/baderyo_o/status/1653780920871055361 If so, could you specify what the advantage of your solution is and what kind of standard you want to implement? Also, for UI-focused work, we usually ask teams to submit some mock-ups or designs together with the application.

Dear @Noc2:
Thanks for reviewing our application! Yes we know Polkadot already has account abstraction and we are pretty excited about it, so our proposal is not about enabling account abstraction in Polkadot. Instead it aims at enabling interoperable and universal third-party account applications among different wallets based on AA. Under the current AA standard, different wallets just develop their own account applications and don't accept third-party account applications developed by others, as a result account applications are fragmented and isolated within each wallet just like Web2 with no interconnectivity and interoperability. Therefore our goal is to make each account application universal and applicable for different AA wallets, and third-party developers can provide their account applications to different wallets.

Furthermore, from the end user side, based on account abstraction, we upgrade it to demand abstraction. Users don't need to look for any specific applications first to meet their demands. Alternatively they can directly tell what they need, and our protocol will analyze user demand intelligently, generate the best solution scheme automatically, match and run the corresponding applications to finally give users the content or services they need from applications.

The link to our [prototype](https://www.figma.com/file/4Yu3MRAu1EG0kQMInFmnJk/Smart-AA-Wallet-Prototype?type=design&node-id=0%3A1&mode=design&t=WJmYKNbNmoL5HPNA-1) has been attached within the Development Status section, feel free to have a look and thanks a lot!

**Review by Noc2 — CHANGES_REQUESTED** [review]
Thanks for the quick reply. But could you specify what exactly you want to develop as part of the milestones? For example, what functionality will the ink! smart contracts provide and how is this different from the proxy pallet? If you leverage the proxy pallet, it also already works with different wallets and is universal in this sense.

**Comment by Noc2** [comment]
I wanted to let you know that I'm closing this because of inactivity. Let me know if I should reopen it.

**Comment by mathida520** [comment]
Dear @Noc2:
Yes we are still working on our proposal and preparing for the rely to your inquiry. I previously assigned our core technical developer Myron Zhang to respond as he knows the technical development on Polkadot via ink!. But recently he is on a business trip unable to rely quickly. I have asked him and he said he would reply soon by the end of this weekend, so could you please reopen our application for him to reply? Thanks a lot for your kindly help!

File: Enabling Third-party Smart Contract Account Applications on Polkadot Based on Account Abstraction.md [hunk 1]
```diff
@@ -0,0 +1,154 @@
...
+| **0c.** | Testing and Testing Guide | Core functions will be fully covered by comprehensive unit tests to ensure functionality and robustness. In the guide, we will describe how to run these tests to verify our AA standard can fully support third-party account applications. |
+| **0d.** | Docker | We will provide Dockerfiles that can be used to test all the functionality delivered with this milestone. |
+| 0e. | Article | We will publish an article that explains how our AA standard works and runs for developers to smoothly provide users with different account applications in their AA accounts. |
+| 1. | User interface | We will design and develop a sample AA wallet for users to have their AA account and equip it with third-party account applications. |
+| 2. | Smart contracts | We will deliver a set of ink! smart contracts to show the specifications of our AA standard supporting third-party account applications. |
+
+
+### Milestone 2 Example — Use Cases of Account applications
```
Comment (semuelle) [inline Enabling Third-party Smart Contract Account Applications on Polkadot Based on Account Abstraction.md:116]:
```suggestion
### Milestone 2 — Use Cases of Account applications
```

File: Enabling Third-party Smart Contract Account Applications on Polkadot Based on Account Abstraction.md [hunk 2]
```diff
@@ -0,0 +1,154 @@
...
+| **0c.** | Testing and Testing Guide | Core functions will be fully covered by comprehensive unit tests to ensure functionality and robustness. In the guide, we will describe how to run these tests to verify the functionalities and security of our use cases.  |
+| **0d.** | Docker | We will provide Dockerfiles that can be used to test all the functionality delivered with this milestone. |
+| 0e. | Article | We will publish an article that explains how our use cases work and run based on our AA standard to encourage more developers to develop their third-party account applications. |
+| 1. | User interface | We will design and develop two use cases for users to use third-party account applications via our AA standard. One is decentralized universal login as an account management application, and the other one is a decentralized automatic trading bot as an account interaction application. |
+| 2. | Smart contracts | We will deliver a set of ink! smart contracts to show the specifications of our third-party account applications as the use cases of our AA standard supporting any third-party account applications. |
+
+
+### Milestone 2 Example — Demand Abstraction
```
Comment (semuelle) [inline Enabling Third-party Smart Contract Account Applications on Polkadot Based on Account Abstraction.md:133]:
```suggestion
### Milestone 3 — Demand Abstraction
```

**Review by semuelle — COMMENTED** [review]
I have reopened the pull request. I will continue reviewing the application once it's updated.

**Comment by mathida520** [comment]
Dear @semuelle:
Hello!

Thanks for your kindly help! We have specified the substrate modules we could develop and included them in each milestone via the new changes we committed. Feel free to review and ask any questions if you have. If that works with no additional questions or critiques, feel free to proceed our applications and review other parts, and we will respond and revise accordingly.

Thanks a lot!

File: Enabling Third-party Smart Contract Account Applications on Polkadot Based on Account Abstraction.md (hunk 2 above)
Comment (mryings) [reply Enabling Third-party Smart Contract Account Applications on Polkadot Based on Account Abstraction.md]: Thanks for your critiques and we have revised them!

File: Enabling Third-party Smart Contract Account Applications on Polkadot Based on Account Abstraction.md (hunk 1 above)
Comment (mryings) [reply Enabling Third-party Smart Contract Account Applications on Polkadot Based on Account Abstraction.md]: Thanks for your critiques and we have revised them!

File: Enabling Third-party Smart Contract Account Applications on Polkadot Based on Account Abstraction.md [hunk 3]
```diff
@@ -0,0 +1,159 @@
...
+- **Payment Address:** bc1quf6qljy5z07sxlv6fgv04zqzaq0g30z3v47v3u (BTC)
+- **[Level](https://github.com/w3f/Grants-Program/tree/master#level_slider-levels):** 2
+
+## Project Overview :page_facing_up:
+
+### Overview
+
+Please provide the following:
+
+The capacity, usability, and personalization of blockchain are limited by the existing crypto account system like externally-owned accounts (EOA) and contract accounts (CA). To address this issue, the recent significant account upgrade called account abstraction (AA) enables users to freely self-define the functions, logic, and settings of their crypto accounts via smart contracts. Compared with the existing EOA and CA, with AA users can have more account functionalities, higher customization, and better usage experience as good as Web2 while maintaining the decentralization of Web3.
```
Comment (semuelle) [inline Enabling Third-party Smart Contract Account Applications on Polkadot Based on Account Abstraction.md:13]: What is this recent significant account upgrade? This sounds like an Ethereum thing.

File: Enabling Third-party Smart Contract Account Applications on Polkadot Based on Account Abstraction.md [hunk 4]
```diff
@@ -0,0 +1,159 @@
...
+
+Furthermore, with account applications, we can further transform account abstraction into the ultimate abstraction which is user demand abstraction. Users can just directly express and meet their demands without caring about the specific achievement across accounts, DApps, and blockchains. In order to do so, our protocol can support AA account applications to intelligently analyze user demands and find the according solution scheme, then match the corresponding DApps and automatically run them on DApps and blockchains for users. Such a better new way of Web3 user interaction based on user demands will be more natural, efficient, and user-friendly than the existing Web2 way of user interaction based on specific apps.
+
+Given the grant, the expected outcome of our project could be:
+
+- A technical standard achieving third-party account applications based on AA
```
Comment (semuelle) [inline Enabling Third-party Smart Contract Account Applications on Polkadot Based on Account Abstraction.md:29]: What do you mean by a "technical standard"?

File: Enabling Third-party Smart Contract Account Applications on Polkadot Based on Account Abstraction.md [hunk 5]
```diff
@@ -0,0 +1,159 @@
...
+| **0d.** | Docker | We will provide Dockerfiles that can be used to test all the functionality delivered with this milestone. |
+| 0e. | Article | We will publish an article that explains how our AA standard works and runs for developers to smoothly provide users with different account applications in their AA accounts. |
+| 1. | User interface | We will design and develop a sample AA wallet for users to have their AA account and equip it with third-party account applications. |
+| 2. | Smart contracts | We will deliver a set of ink! smart contracts to show the specifications of our AA standard supporting third-party account applications. |
+| 3. | Substrate module: transaction execution | We will create a Substrate module to define and achieve the interfaces of executing transactions. |
+| 4. | Substrate module: identity verification | We will create a Substrate module to define and achieve the interfaces of verifying the identity of users. |
```
Comment (semuelle) [inline Enabling Third-party Smart Contract Account Applications on Polkadot Based on Account Abstraction.md:115]: What's the purpose of identifying the identity of users? What identity? What do you mean by "achieve the interfaces of executing transactions"? How do the smart contracts and pallets interact in this architecture?

**Review by semuelle — CHANGES_REQUESTED** [review]
Hi @mathida520, thanks for the updates. I left a couple of questions below. But my main issue is that I simply don't understand what a lot of the phrases you are using refers to. What are account applications? What are AA wallets? 

Also, isn't all of this already possible with proxy accounts? Can you provide an architecture diagram to explain the problem you are trying to solve? Perhaps describe an example problem. I would recommend to leave the "intelligent analysis" out of it for now. There is a section in our application template that should describe projects/technologies that are similar to what is being proposed. It would be great if you could explain what other solutions might exist and how they fall short, whether it's in the Polkadot ecosystem or other ecosystems.

Lastly, does anyone on your team have experience with Substrate, or building and maintaing open source software? I couldn't find much under the accounts you linked.

**Comment by semuelle** [comment]
pinging @mathida520

**Comment by semuelle** [comment]
I am closing this application due to inactivity. Please ping us here, @mathida520, if you would like to continue working on it.

**Pull request closed without merging**